In [25]:
import networkx as nx
import matplotlib.pyplot as plt
%matplotlib inline

In [8]:
M=nx.read_gml("polblogs.gml")

In [9]:
# create weighted graph from M
G = nx.Graph()
for u,v,data in M.edges_iter(data=True):
    if G.has_edge(u,v):
        G[u][v]['weight'] += 1
    else:
        G.add_edge(u, v, weight=1)



In [20]:
centrality= nx.eigenvector_centrality_numpy(G)

In [28]:
blog_list = []
for node in centrality:
    blog_list.append((node,centrality[node]))
    
sorted_blog_list = blog_list.sort(key=lambda x: x[1])

In [29]:
sorted(blog_list, key=lambda x: x[1], reverse=True)

[(u'atrios.blogspot.com', 0.19482295898856763),
 (u'dailykos.com', 0.16090716385818687),
 (u'washingtonmonthly.com', 0.14654050463836452),
 (u'digbysblog.blogspot.com', 0.14265582251978648),
 (u'bodyandsoul.typepad.com', 0.14114853531288896),
 (u'liberaloasis.com', 0.1395378566665858),
 (u'tbogg.blogspot.com', 0.1370485753554684),
 (u'corrente.blogspot.com', 0.12943073134224975),
 (u'talkingpointsmemo.com', 0.1270093107829415),
 (u'dneiwert.blogspot.com', 0.12238703900637997),
 (u'pandagon.net', 0.12184646040285349),
 (u'billmon.org', 0.11750626591024393),
 (u'xnerg.blogspot.com', 0.11480900917867652),
 (u'madkane.com/notable.html', 0.11419082729577247),
 (u'elayneriggs.blogspot.com', 0.11265673556127069),
 (u'busybusybusy.com', 0.11025501978461205),
 (u'michaelberube.com', 0.10609151756254845),
 (u'pacificviews.org', 0.10578482549977733),
 (u'nielsenhayden.com/electrolite', 0.10553449257446292),
 (u'instapundit.com', 0.10538942033754214),
 (u'prospect.org/weblog', 0.10434664082865737)

In [31]:
sorted(blog_list, key=lambda x: x[1], reverse=False)

[(u'thelonedem.com', -3.8587743106527666e-19),
 (u'digital-democrat.blogspot.com', 1.8546566127753985e-18),
 (u'americanworldview.tripod.com/weltansblog', 7.489792406515822e-09),
 (u'quimundus.modblog.com', 6.675725344883722e-08),
 (u'batr.org/commentary.html', 6.794245454788024e-07),
 (u'batr.net', 6.795053204372692e-07),
 (u'democratvoice.org', 1.0221602317205466e-06),
 (u'quimundus.squarespace.com', 6.0564974874421066e-06),
 (u'massachusetts-liberal.com', 9.484711746615202e-06),
 (u'thingsasitis.net', 1.1737028626602236e-05),
 (u'loveamericahatebush.com', 2.0343745233533475e-05),
 (u'imprescindibile.ilcannocchiale.it', 2.3891392483796096e-05),
 (u'eltcamerica.blogspot.com', 2.836339841891837e-05),
 (u'usademocrazy.blogspot.com', 2.984877356285562e-05),
 (u'depressedlonelybored.typepad.com', 3.037526214597933e-05),
 (u'margieburns.com', 3.32030272676951e-05),
 (u'endthenightmare.blogspot.com', 3.5887337468882995e-05),
 (u'amliberal.com/blog', 3.621782064231649e-05),
 (u'mcwil.blogspo

### 1.  Create a nice looking graph of the political blog network

In [ ]:
plt.figure(figsize=(80,80))
pos=nx.spring_layout(G, iterations=500, scale=5, k=7)
nx.draw_networkx_labels(G,pos,fontsize=8)
font = {'fontname'   : 'Helvetica',
            'color'      : 'k',
            'fontweight' : 'bold',
            'fontsize'   : 14}
plt.title("Blogs", font)


nx.draw_networkx_nodes(G,pos,node_color='b',alpha=0.4)
nx.draw_networkx_edges(G,pos,alpha=0.4,node_size=0,width=1,edge_color='k')
plt.axis('off')
plt.show()

### 2.  What are the 10 most influential blogs?

 (u'atrios.blogspot.com', 0.19482295898856763),
 (u'dailykos.com', 0.16090716385818687),
 (u'washingtonmonthly.com', 0.14654050463836452),
 (u'digbysblog.blogspot.com', 0.14265582251978648),
 (u'bodyandsoul.typepad.com', 0.14114853531288896)

### 3.  What are the 10 least influential blogs?



 (u'thelonedem.com', -3.8587743106527666e-19),
 (u'digital-democrat.blogspot.com', 1.8546566127753985e-18),
 (u'americanworldview.tripod.com/weltansblog', 7.489792406515822e-09),
 (u'quimundus.modblog.com', 6.675725344883722e-08),
 (u'batr.org/commentary.html', 6.794245454788024e-07)

### 4.  Answer this question for both left wing and right wing blogs seperately.  (Optional 10 Extra Credit Points)